### ISSUE TITLE
[visually-hidden] attr

### ISSUE THREAD

**Issue by thepassle** [thepassle/generic-components#13]
As discussed with Brian, it would make a lot of sense to have visually-hidden be an attribute rather than a web component. Something like:

```html
<p visually-hidden>I am visually hidden!</p>
```

Could be easily achieved by shipping some CSS like:

```css
[visually-hidden] {
  clip: rect(1px, 1px, 1px, 1px);
  clip-path: inset(50%);
  height: 1px;
  width: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
}
```

My main concern with this is that this wouldnt work if your app uses a lot of shadow DOM. E.g., if I'm developing an app with web components:

```js
class MyEl extends HTMLElement {
  constructor() {
    super();
    this.attachShadow({ mode: 'open' });
    this.shadowRoot.innerHTML = `<p visually-hidden>I am visually hidden!</p>`
  }
}
```

The `[visually-hidden]` selector wouldn't reach this paragraph because its in a shadowroot. Once module attributes/css modules lands, that could be solved like:

```js
import visuallyHidden from './visually-hidden.css';

class MyEl extends HTMLElement {
  constructor() {
    super();
    this.attachShadow({ mode: 'open' });
    this.shadowRoot.adoptedStyleSheets = [visuallyHidden]
  }
}
```

But even if that lands, it'll be a long time before they'll be widely supported.

An option here is to ship both a component, and a css 'util' for the time being, although that may be confusing to a user.

**Comment by thepassle** [comment]
oh lol, duplicate of #7
